elytron/ELY-1686-legacy-properties-files-into-FileSystemRealm-utility.adoc

```diff
@@ -0,0 +1,199 @@
+It should be possible to specify the password using any supported mechanism.
+
+** Command with only required options
+[source,bash]
+./bin/elytron-tool.sh filesystem-realm --user-file/-u <userFile> --roles-file/-r <rolesFile>
+--output-location/-o <outDirectory> --password/-p <password>
+
+** Command with all possible options
+[source,bash]
+./bin/elytron-tool.sh filesystem-realm --user-file/-u <userFile> --roles-file/-r <rolesFile>
+--output-location/-o <outDirectory> --password/-p <password> [--filesystem-realm/-f] <name>
+[--security-domain/-s] <name> [--http-authentication-factory/-h] <name>
+[--application-security-domain/-a] <name>
+
+* There should be a text descriptor file that allows multiple inputs to be specified. The
+descriptor file should only accept the long-form option names for readability. Using the
+text descriptor file will have its own option, --bulk-convert/-b
+
+** CLI Command
+[source,bash]
+./bin/elytron-tool.sh filesystem-realm --bulk-convert/-b <file>
+
+** Text descriptor file
+[source,text]
+----
+user-file:/relative/path/to/userFile1
+roles-file:/relative/path/to/rolesFile1
+output-location:/relative/path/to/output/outDirectory1
+password:examplePasswordClearText1
+filesystem-realm:nameOfFileSystemRealm1
+security-domain:nameOfSecurityDomain1
+http-authentication-factory:nameOfHttpAuthenticationFactory1
+application-security-domain:nameOfApplicationSecurityDomain1
+
+user-file:/absolute/path/to/userFile2
+roles-file:/absolute/path/to/rolesFile2
+output-location:/absolute/path/to/output/outDirectory2
+password:examplePasswordOtherMechanism2
+filesystem-realm:nameOfFileSystemRealm2
+security-domain:nameOfSecurityDomain2
+http-authentication-factory:nameOfHttpAuthenticationFactory2
+application-security-domain:nameOfApplicationSecurityDomain2
+----
+
+* Summary of options and their description
+|===
+|Long-form |Short-form |Description
+
+| --user-file
+| -u
+| The relative or absolute path to the users file
+
+| --roles-file
+| -r
+| The relative or absolute path to the roles file
+
+| --output-location
+| -o
+| The relative or absolute path to the output directory
+
+| --password
+| -p
+| The password, given as in the form of a supported mechanism
+
+| --filesystem-realm
+| -f
+| Name of the filesystem-realm to be generated
+
+| --security-domain
+| -s
+| Name of the security-domain to be generated
+
+| --http-authentication-factory
+| -h
+| Name of the http-authentication-factory to be generated
+
+| --application-security-domain
+| -a
+| Name of the application-security-domain to be generated
+
+| --bulk-convert
+| -b
+| Tells the command to use a descriptor file, for multiple users-roles file combinations
+can be provided.
+|===
+
+=== Nice-to-Have Requirements
+
+* The command should have an interactive option, similar to the -i option in the unix rm
+command, where the user is prompted for confirmation prior to each file conversion
```

Comment:
Since the user is explicitly specifying the files they want to convert in the descriptor file, I wonder if it makes sense to still prompt them for each conversion. May be good to check the behaviour of the existing Vault command to see what it does since its bulk conversion process is similar.